### DISCUSSION TITLE
Reevaluate filtered query result based on some changed data

### DISCUSSION THREAD

**Discussion by hognevevle** [urql-graphql/urql#1672] [Q&A] [answered]
Hi guys!

I'm trying to find out how I can most easily achieve real time updates in my app using subscriptions. The current challenge is that I have different views based on filters. Let's consider an example of an email inbox, which is similar to my scenario. where each email can be unread, read, deleted, archived etc..

Each property change to an email will emit a message through a subscription (e.g. a `onEmailUpdated` subscription). Thanks to the urql cache, it will automagically detect all these changes and update the UI, _however_, it does not reevaluate the query results in relation to the applied filter. For example, let's say my query has a filter `unread: {eq: true}` , and then an email gets read after the fact. The `seen` state will update in my UI (again thanks to the urql cache), but the email will still be present in the query results, even though it does not match the filter anymore. 

Are there any easy ways to deal with this scenario?

**Comment by kitten** [comment, marked as answer]
Hiya :wave:

I believe you're correct on all points. If you have subscriptions set up then with Graphcache the normalised cache structure will update any matching entities, but lists will be left out of that as a cache can't assme how lists will update in reaction to changing messages.

In these cases you'd need to write manual cache updates:

- https://formidable.com/open-source/urql/docs/graphcache/normalized-caching/#manual-cache-updates
- https://formidable.com/open-source/urql/docs/graphcache/cache-updates/

This would require you to write some small pieces of logic that update your lists to exclude changing items depending on their updated properties.

**Reply by hognevevle** [reply]
Thank you!
